pkg/cvo/cvo_scenarios_test.go

```diff
@@ -1566,6 +1566,233 @@ func TestCVO_UpgradeUnverifiedPayload(t *testing.T) {
+	retriever.Set(PayloadInfo{}, payloadErr)
+
+	go worker.Start(ctx, 1, o.name, o.cvLister)
+
+	// Step 1: The operator should report that it is blocked on unverified content
+	//
+	client.ClearActions()
+	err := o.sync(ctx, o.queueKey())
+	if err != nil {
+		t.Fatal(err)
+	}
+	actions := client.Actions()
+	verifyCVSingleUpdate(t, actions)
+
+	verifyAllStatus(t, worker.StatusCh(),
+		SyncWorkerStatus{
+			Actual:     configv1.Release{Version: "1.0.1-abc", Image: "image/image:1"},
+			Generation: 1,
+			loadPayloadStatus: LoadPayloadStatus{
+				Step:               "RetrievePayload",
+				Message:            "Retrieving and verifying payload version=\"1.0.1-abc\" image=\"image/image:1\"",
+				LastTransitionTime: time.Unix(1, 0),
+				Release:            configv1.Release{Version: "1.0.1-abc", Image: "image/image:1"},
+			},
+		},
+		SyncWorkerStatus{
+			Actual:       configv1.Release{Version: "1.0.1-abc", Image: "image/image:1"},
+			Generation:   1,
+			LastProgress: time.Unix(1, 0),
+			loadPayloadStatus: LoadPayloadStatus{
+				Step:               "RetrievePayload",
+				Message:            "Retrieving payload failed version=\"1.0.1-abc\" image=\"image/image:1\" failure=The update cannot be verified: some random error",
+				LastTransitionTime: time.Unix(2, 0),
+				Release:            configv1.Release{Version: "1.0.1-abc", Image: "image/image:1"},
+				Failure:            payloadErr,
+			},
+		},
+	)
+	actions = client.Actions()
+	if len(actions) != 2 {
+		t.Fatalf("%s", spew.Sdump(actions))
+	}
+	expectGet(t, actions[0], "clusterversions", "", "version")
+	expectUpdateStatus(t, actions[1], "clusterversions", "", &configv1.ClusterVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "version",
+			ResourceVersion: "1",
+			Generation:      1,
+		},
+		Spec: configv1.ClusterVersionSpec{
+			ClusterID:     clusterUID,
+			Channel:       "fast",
+			DesiredUpdate: &configv1.Update{Version: desired.Version, Image: desired.Image},
+			Capabilities: &configv1.ClusterVersionCapabilitiesSpec{
+				BaselineCapabilitySet:         configv1.ClusterVersionCapabilitySetNone,
+				AdditionalEnabledCapabilities: []configv1.ClusterVersionCapability{"baremetal"},
+			},
+		},
+		Status: configv1.ClusterVersionStatus{
+			// Prefers the image version over the operator's version (although in general they will remain in sync)
+			ObservedGeneration: 1,
+			Desired:            desired,
+			VersionHash:        "DL-FFQ2Uem8=",
+			History: []configv1.UpdateHistory{
+				{State: configv1.PartialUpdate, Image: "image/image:1", Version: "1.0.1-abc", StartedTime: defaultStartedTime},
+				{State: configv1.CompletedUpdate, Image: "image/image:0", Version: "1.0.0-abc", Verified: true, StartedTime: defaultStartedTime, CompletionTime: &defaultCompletionTime},
+			},
+			Conditions: []configv1.ClusterOperatorStatusCondition{
+				{Type: ImplicitlyEnabledCapabilities, Status: "False", Reason: "AsExpected", Message: "Capabilities match configured spec"},
```

Comment:
I expected `status.Capabilities` to be set at this point.  Perhaps that is not something that `o.sync` syncs?

Reply:
Right. I changed `Update` to also set CV status before returning on load error.